components/contribution-flow/StepDetailsCrypto.js

```diff
@@ -71,30 +51,27 @@ const validateCryptoCurrencyAmount = (touched, amount, minimumAmount, cryptoCurr
 
 const StepDetailsCrypto = ({ onChange, data, collective }) => {
   const intl = useIntl();
+
   const [selectedCryptoCurrency, setSelectedCryptoCurrency] = useState(data.currency);
   const [amount, setAmount] = useState(data.amount);
   const [convertedAmount, setConvertedAmount] = useState(null);
-  const [cryptoExchangeRate, setCryptoExchangeRate] = useState(null);
   const [touched, setTouched] = useState(false);
   const collectiveCurrency = collective.currency;
   const dispatchChange = (field, value) => {
     onChange({ stepDetails: { ...data, [field]: value }, stepSummary: null });
   };
 
-  const storeCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
-    const exchangeRate = await getCryptoExchangeRate(cryptoCurrency, collectiveCurrency);
-    setCryptoExchangeRate(exchangeRate);
-  };
+  const { data: fxData } = useQuery(cryptoExchangeRateQuery, {
+    context: API_V2_CONTEXT,
+    variables: { cryptoCurrency: selectedCryptoCurrency.value, collectiveCurrency },
+    skip: !selectedCryptoCurrency,
+  });
 
   useEffect(() => {
-    storeCryptoExchangeRate(selectedCryptoCurrency.value, collectiveCurrency);
-  }, []);
-
-  useEffect(() => {
-    if (cryptoExchangeRate) {
-      setConvertedAmount(amount * cryptoExchangeRate);
+    if (fxData) {
+      setConvertedAmount(amount * fxData.cryptoExchangeRate);
     }
-  }, [amount, cryptoExchangeRate]);
+  }, [amount, fxData]);
```

Comment:
What kind of failure are you taking care of? What happens if `fxData.cryptoExchangeRate` is null?